cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -1058,40 +715,23 @@ impl MachInstEmit for Inst {
                 from_bits,
                 to_bits: _to_bits,
             } => {
-                todo!() /* let rn = allocs.next(rn);
-                        let rd = allocs.next_writable(rd);
-                        let mut insts = SmallInstVec::new();
-                        let shift_bits = (64 - from_bits) as i16;
-                        let is_u8 = || from_bits == 8 && signed == false;
-                        if is_u8() {
-                            // special for u8.
-                            insts.push(Inst::AluRRImm12 {
-                                alu_op: AluOPRRI::Andi,
-                                rd,
-                                rs: rn,
-                                imm12: Imm12::from_bits(255),
-                            });
-                        } else {
-                            insts.push(Inst::AluRRImm12 {
-                                alu_op: AluOPRRI::Slli,
-                                rd,
-                                rs: rn,
-                                imm12: Imm12::from_bits(shift_bits),
-                            });
-                            insts.push(Inst::AluRRImm12 {
-                                alu_op: if signed {
-                                    AluOPRRI::Srai
-                                } else {
-                                    AluOPRRI::Srli
-                                },
-                                rd,
-                                rs: rd.to_reg(),
-                                imm12: Imm12::from_bits(shift_bits),
-                            });
-                        }
-                        insts
-                            .into_iter()
-                            .for_each(|i| i.emit(&[], sink, emit_info, state)); */
+                let rn = allocs.next(rn);
+                let rd = allocs.next_writable(rd);
+                assert_eq!(rn, b0());
+                assert_eq!(rd.to_reg(), a0());
+                assert_eq!(from_bits, 32);
+                assert_eq!(_to_bits, 64);
+                Inst::Mov {
+                    rd,
+                    rm: rn,
+                    ty: I64,
+                }
+                .emit(&[], sink, emit_info, state);
+                if signed {
+                    put_string(&format!("0x80000000n => B\n"), sink);
+                    put_string(&format!("$ => A: XOR\n"), sink);
+                    put_string(&format!("$ => A: SUB\n"), sink);
```

Comment:
Fancy.

Reply:
Yeah, generating a code with branching was surprisingly hard here, so I just avoided it 🙈